wiki/C++-Style-Guide.md

```diff
@@ -0,0 +1,814 @@
+# Table of Contents
+
+* [Background](#background)
+* [Files](#files)
+* [Namespaces](#namespaces)
+* [Classes](#classes)
+* [Functions](#functions)
+* [Variables](#variables)
+* [Comments](#comments)
+* [Formatting](#formatting)
+* [Best practices](#best-practices)
+* [C++11](#c11)
+
+# Background
+
+This style guide is also available on the [website](https://endless-sky.github.io/styleguide/styleguide.xml).
+
+Each section of this style guide can be read as a "how to": how to create source code files, how to create classes, etc. The sections are ordered hierarchically, from biggest "units" (files, namespaces, classes) to increasingly smaller ones (variables, comments, formatting). Our style choices are based on a small set of principles:
+
+- Readability is achieved by laying out text in a clear visual hierarchy.
+- Code should not rely on features of a particular editor or compiler.
+- Formatting should be consistent to avoid confusing or distracting the reader.
+- Reducing the risk of bugs is important even if it takes more work.
+
+
+# Files
+
+This section deals with naming and encoding of source code files and the items that appear at the start of each file.
+
+## Line endings
+
+All files should have Unix-style line endings.
+That is, "\n" alone, not "\r\n" (Windows style). If an editor changes Unix line endings to Windows style, this causes issues with source control diffs because those lines appear to be changed even though their actual text is the same.
+
+## File encodings
+
+All files should be plain ASCII.
+That is, nothing beyond the 127-character ASCII set should be used. This is to ensure maximum cross-compatibility.
+
+## File extensions
+
+All C++ files should use ".h" and ".cpp" for their extensions.
+This is a somewhat arbitrary choice, but these seem to be the most commonly used extensions.
+
+## One class per file
+
+Each class "MyClass" should have its own .h and .cpp files, and they should be named "MyClass.h" and "MyClass.cpp".
+This is so that it is easy to find the implementation of any class you are interested in.
+
+## Names for files with no classes
+
+If a file does not contain any classes (such as "main.cpp" or "shift.h"), its name should be lower-case.
+This is just to make it clear that the file is not defining a class.
+
+## Header file guards
+
+All header files should have #define guards, and they should be in the format "#ifndef MY_CLASS_H_".
```

Comment:
```suggestion
All header files should have #define guards, and they should be in the format `#ifndef MY_CLASS_H_`.
```

Reply:
I need to update lines about header guards anyway, so this will be removed.